tensorflow/python/util/nest.py

```diff
@@ -260,8 +260,9 @@ def flatten(structure, expand_composites=False):
   running.
 
   Args:
-    structure: an arbitrarily nested structure or a scalar object. Note, numpy
-      arrays are considered scalars.
+    structure: an arbitrarily nested structure which can be a scalar, or
+      tuple or dict or list of constructed scalars and/or other tuples/lists, or
+      a scalar object. Note, numpy arrays are considered scalars.
```

Comment:
Suggestion: "an arbitrarily nested structure of list, tuple, namedtuple and dicts. Note, numpy arrays are considered atoms and are not flattened."

Could you also update the text at line 246?

Alternatively, you can define what a structure is in the module overview and refer to that from everywhere else, to avoid the duplication.

Reply:
I'm thinking about using the word "atoms" instead of "scalars" to avoid confusion with scalar tensors (these structures can contain anything)

Reply:
> Alternatively, you can define what a structure is in the module overview and refer to that from everywhere else, to avoid the duplication.

This is already done on line 18